syntax = "proto3";
package minigame_responses;

message SampleResponse {
  enum Role {
    kFirst = 0;
    kSecond = 1;
    kThird = 2;
    kFourth = 3;
  }
  map<uint64, string> answers = 1;
  Role role = 2;
}

message InitialTerminalResponse {
  repeated uint64 available_buttons_id = 1;
  string text = 2;
}

message TerminalResponse {
  string entered_text = 1;
}

message MinigameResponse {
  enum Result {
    kCompleted = 0;
    kFailed = 1;
  }
  uint64 remaining_time = 1;
  oneof type {
    Result result = 2;
    SampleResponse sample_response = 3;
    InitialTerminalResponse initial_terminal_respone = 4;
    TerminalResponse terminal_response = 5;
  }
}